### PR TITLE
Stop Departure Ordering

### LINKED ISSUE #246: Display stop departures per route per direction
Like businfoboard (so, for example, OH and MC buses show up separately)

https://umts.github.io/pvta-multiplatform/#/app/stops/58
http://umts.github.io/BusInfoBoard/?stops=58

multiplatform-stop58

businfoboard-stop58

### PR DESCRIPTION
**Pull request by akaplo**
Closes #246.  Sorts departures by Route Direction by default (thus, for example, allowing both a section of Mullins Center via CS departures **and** a section of OHill via Sylvan departures).  It also adds the option to sort departures by time.

There's a ton of logic changes (and even some simplification!) here.  @petermarathas and @bgregg, please review code and UI.  @sherson and @werebus, please give your opinions as well.

**CURRENT VIEW**
image

**NEW VIEW (sorting by route direction)**
image

**NEW VIEW (sorting by time; they run off the page as there's about 20)**
image

### CODE REVIEW

**Comment by sherson** [comment]
UI looks good.

**Review by petermarathas — CHANGES_REQUESTED** [review]
Looks good, but what if we stored the users preference (time vs. direction)? So if the person prefers sorting by time, they wouldn't need to change it each time they opened a stop.

Something like localforage.set({directionPreference: 1}) in the sort method

File: www/pages/stop/stop-controller.js [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
   ga('send', 'pageview');
+  $scope.ROUTE_DIRECTION = '0';
+  $scope.TIME = '1';
+  $scope.filterOptions = ['Route', 'Time'];
+  $scope.sort = $scope.ROUTE_DIRECTION;
+
+  /**
+   * Lets the user choose how they want departures to be sorted
+   */
+  $scope.chooseFilter = function () {
```
Comment (petermarathas) [inline www/pages/stop/stop-controller.js]: Is this method used? Can't find it anywhere in the html or controller

**Comment by akaplo** [comment]
@petermarathas implemented your comment, see [these lines](https://github.com/umts/pvta-multiplatform/pull/256/files#diff-9dc3af022ce52fcaff8bc239047fab0cR198).  Let me know if there's anything else.

**Comment by akaplo** [comment]
@bgregg this is a big update, I'd like your opinion on the code if possible.

**Comment by akaplo** [comment]
CC failures are OK.

**Comment by bgregg** [comment]
Looking now, please hold

File: www/pages/stop/stop-controller.js [hunk 2]
```diff
@@ -188,26 +194,48 @@ angular.module('pvta.controllers').controller('StopController', function ($scope
...
+    var funcName = 'StopController.saveOrdering()';
+    if ($scope.order === $scope.ROUTE_DIRECTION) {
+      ga('send', 'event', eventName, funcName, 'By Route Direction');
+    }
+    else {
+      ga('send', 'event', eventName, funcName, 'By Time');
+    }
+  }
+
+  function loadOrdering () {
```
Comment (petermarathas) [inline www/pages/stop/stop-controller.js:210]: 👍

**Review by petermarathas — APPROVED** [review]

**Review by bgregg — APPROVED** [review]
👍

**Pull request merged**